File: src/main/java/net/dashflight/data/postgres/PostgresConnectionOptions.java
package net.dashflight.data.postgres;

import lombok.Builder;
import lombok.Data;

/**
 * Holds connection data for connecting to Postgres instance
 */
@Data
@Builder
public class PostgresConnectionOptions {

    /**
     * The host address of the instance
     */
    private String host;

    /**
     * The port of the instance
     */
    private int port;

    /**
     * The name of the database to connect to
     */
    private String dbname;

    /**
     * The username to authenticate with
     */
    private String username;

    /**
     * The password to authenticate with
     */
    private String password;

    /**
     * The name of the application connecting. This name shows up under `pg_stat_activity` table
     */
    @Builder.Default
    private String applicationName = "Java Application";

    /**
     * Maximum size of the connection pool
     */
    @Builder.Default
    private int maxPoolSize = 2;
}

File: src/main/java/net/dashflight/data/postgres/PostgresConnectionOptions.kt
package net.dashflight.data.postgres

/**
 * Holds connection data for connecting to Postgres instance
 */
data class PostgresConnectionOptions(val host: String, val port: Int, val dbname: String?, val username: String?, val password: String?, val applicationName: String = "Java Application", val maxPoolSize: Int = 2) {
    /**
     * The host address of the instance
     */

    /**
     * The port of the instance
     */

    /**
     * The name of the database to connect to
     */

    /**
     * The username to authenticate with
     */

    /**
     * The password to authenticate with
     */

    /**
     * The name of the application connecting. This name shows up under `pg_stat_activity` table
     */

    /**
     * Maximum size of the connection pool
     */
}